game-app/game-core/src/main/java/games/strategy/triplea/ui/BattleDisplay.java

```diff
@@ -533,87 +525,39 @@ CasualtyDetails getCasualties(
-                    chooserScrollPane = new JScrollPane(chooser);
-                    final Dimension screenResolution = Toolkit.getDefaultToolkit().getScreenSize();
-                    final int availHeight = screenResolution.height - 130;
-                    final int availWidth = screenResolution.width - 30;
-                    chooserScrollPane.setBorder(BorderFactory.createEmptyBorder(1, 1, 1, 1));
-                    final Dimension size = chooserScrollPane.getPreferredSize();
-                    chooserScrollPane.setPreferredSize(
-                        new Dimension(
-                            Math.min(
-                                availWidth,
-                                size.width
-                                    + (size.height > availHeight
-                                        ? chooserScrollPane
-                                            .getVerticalScrollBar()
-                                            .getPreferredSize()
-                                            .width
-                                        : 0)),
-                            Math.min(availHeight, size.height)));
-                  }
-                  final String[] options = {"Ok", "Cancel"};
-                  final String focus =
-                      ClientSetting.spaceBarConfirmsCasualties.getValueOrThrow()
-                          ? options[0]
-                          : null;
-                  final int option =
-                      JOptionPane.showOptionDialog(
-                          BattleDisplay.this,
-                          chooserScrollPane,
-                          hit.getName() + " select casualties",
-                          JOptionPane.YES_NO_OPTION,
-                          JOptionPane.PLAIN_MESSAGE,
-                          null,
-                          options,
-                          focus);
-                  if (option != 0) {
-                    actionButton.setEnabled(true);
-                    return;
-                  }
-                  final List<Unit> killed = chooser.getSelected(false);
-                  final List<Unit> damaged = chooser.getSelectedDamagedMultipleHitPointUnits();
-                  if (!isEditMode && (killed.size() + damaged.size() != count)) {
-                    JOptionPane.showMessageDialog(
-                        BattleDisplay.this,
-                        "Wrong number of casualties selected",
-                        hit.getName() + " select casualties",
-                        JOptionPane.ERROR_MESSAGE);
-                  } else {
-                    final CasualtyDetails response = new CasualtyDetails(killed, damaged, false);
-                    casualtyDetails.set(response);
-                    dicePanel.removeAll();
-                    actionButton.setAction(nullAction);
-                    continueLatch.countDown();
-                  }
-                  actionButton.setEnabled(true);
+            new AbstractAction(btnText) {
```

Comment:
Were you able to find a way to remove this code, or is it moved somewhere that I do not see?

Are we able to remove all this code readily without replacing it? 

I suspect there is a problem introduced here. I tried the casualty selector and selected the wrong number of casualties and the game did not re-prompt for the correct number of casualties.

Reply:
Will be fixed with the next commit. Three new UI test cases to prove the fix won't be committed, but await our discussion once this PR is done ;)

Reply:
Wait, I experimented with this branch and selected the incorrect number of casualties. The game became stuck because the retry code that was removed here is no longer there. The game did not prompt me to select the correct number of casualties. If we merge as-is, it would be a major problem. We should not put the code into an unreleasable state willingly.

Reply:
With the currently commited version I tested both isolated and within 240BC. After I read your comment abve, I also tested in Big World: 1942, World War II Classic, and World War II Classic 3rd Edition. When I enter the wrong number of casualties, I get the "Wrong number of casualties selected" popup, and the `_player_ select _n_ casualties` button at the bottom of the battle dialog stays active. I can see no deviation from the required behavior. Could you please tell me the game and the steps to the error?

Reply:
I'll check if I can still repro and will let you know.